misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/MockFunctionReplaceVisitor.java

```diff
@@ -61,41 +58,23 @@ public MockFunctionReplaceVisitor(int api, ClassWriter cw, String name, String m
 
     @Override
     public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
-
+        MethodVisitor defaultMethodVisitor;
         if (!name.equals(methodName) || !desc.equals(methodDesc)) {
             // reproduce the methods where no changes needed
-            return super.visitMethod(access, name, desc, signature, exceptions);
-        }
-
-        return new MethodReplaceAdapter(
-                super.visitMethod(access, name, desc, signature, exceptions), mockFunc);
-    }
+            defaultMethodVisitor = super.visitMethod(access, name, desc, signature, exceptions);
+        } else {
+            // Rename function as $ORIG_name.
+            defaultMethodVisitor = super.visitMethod(access, ORIGINAL_FUNC_NAME_PREFIX + name,
+                    desc, signature, exceptions);
 
+            // Create a new method with the same name of original method and call the method $MOCK_name inside it.
+            MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
```

Comment:
Shall we move duplicate method creation logic tos eperate method

Reply:
Addressed with the commit https://github.com/ballerina-platform/ballerina-lang/pull/36115/commits/d7ac7b336c1cc3f8d8eba078c20f21ebc279088c